hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/operation/EntrySetOperation.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.replicatedmap.impl.operation;
+
+import com.hazelcast.internal.serialization.SerializationService;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.replicatedmap.impl.ReplicatedMapService;
+import com.hazelcast.replicatedmap.impl.client.ReplicatedMapEntrySet;
+import com.hazelcast.replicatedmap.impl.record.ReplicatedRecord;
+import com.hazelcast.replicatedmap.impl.record.ReplicatedRecordStore;
+import com.hazelcast.spi.AbstractOperation;
+import java.io.IOException;
+import java.util.AbstractMap;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class EntrySetOperation extends AbstractOperation {
+
+    private String name;
+    private transient Object response;
+
+    public EntrySetOperation() {
+    }
+
+    public EntrySetOperation(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public void run() throws Exception {
+        ReplicatedMapService service = getService();
+        Collection<ReplicatedRecordStore> stores = service.getAllReplicatedRecordStores(name);
+        Set<Map.Entry<Object, ReplicatedRecord>> entrySet = new HashSet<Map.Entry<Object, ReplicatedRecord>>();
```

Comment:
We know that these entries are distinct by their key. Then I think we can put them into an array list etc. (some collection does not require hashing) and return it that way to the user. I suppose we did something similar with the previous impl. Something like lazy collection etc. Additionally, I think we can remove one of these ``entrySet`` and ``entries`` and use only one.